### ISSUE TITLE
[telemetry] Login, logout and logged-in-user metrics in LoggedInUserManager

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#43]
## Goal

Emit login and logout metrics from `LoggedInUserManager` in `ph-oton-security`.

**This is the highest-value single instrument in the whole repository.** A failed-login counter broken down by reason is the difference between noticing a credential-stuffing attempt and not noticing it. Today the only trace of a failed login is a log line.

## Where

`ph-oton-security/src/main/java/com/helger/photon/security/login/LoggedInUserManager.java`

**1. Failure choke point** — every single login failure in the class funnels through this one private method:

```java
@NonNull
private ELoginResult _onLoginError (@NonNull @Nonempty final String sUserID, @NonNull final ELoginResult eLoginResult)
{
  m_aUserLoginCallbacks.forEach (aCB -> aCB.onUserLoginError (sUserID, eLoginResult));
  return eLoginResult;
}
```

It is reached from `loginUser (IUser, ...)` for `USER_IS_DELETED`, `USER_IS_DISABLED`, `INVALID_PASSWORD`, `USER_IS_MISSING_ROLE`, `SESSION_ALREADY_HAS_USER`, `USER_ALREADY_LOGGED_IN`, and from `internalSessionActivateUser (...)` for the session-reactivation failures. One counter increment here covers all of them.

**2. Success** — the tail of `loginUser (@Nullable IUser aUser, ...)`, which returns `SUCCESS` or `SUCCESS_WITH_LOGOUT`.

**3. Logout** — `logoutUser (@Nullable String sUserID)`. Note its early-return path when `m_aLoggedInUsers.remove (sUserID)` yields `null` (`EChange.UNCHANGED`, "user-not-logged-in"); that path must not count as a logout.

**4. Concurrent users** — the class already maintains `m_aLoggedInUsers` and exposes `getLoggedInUserCount ()`. Prefer an **observable gauge** over `getLoggedInUserCount ()` rather than a manually balanced up-down counter: the gauge cannot drift, and there are several ways a user leaves the map (explicit logout, session destruction, `UserModificationLogoutCallback`).

`USER_NOT_EXISTING` is returned directly from `loginUser (String sLoginName, ...)` **without** going through `_onLoginError` — it needs its own increment at that site, otherwise the most interesting failure case (user enumeration attempts) is the one that goes unrecorded.

## Cardinality and privacy — hard rules

- `ELoginResult` has exactly 10 values. It is the ideal metric dimension.
- The **user ID must never be a metric attribute** — unbounded, and personal data.
- The user ID on a *span* attribute must respect `isAnonymousLogging ()`, exactly as `_getUserIDLogText (...)` already does for logging. When anonymous logging is on, omit the attribute entirely.
- No login names, no passwords, no IP addresses on metrics. IP-based throttling metrics are covered by the separate `LoginThrottlePerIP` issue.

## What to emit

New `com.helger.photon.security.CSecurityTelemetry`:

| Constant | Value |
|---|---|
| `METRIC_LOGIN_SUCCESS` | `photon.security.login.success` |
| `METRIC_LOGIN_FAILED` | `photon.security.login.failed` |
| `METRIC_LOGOUT` | `photon.security.logout` |
| `METRIC_USERS_LOGGED_IN` | `photon.security.users.loggedin` |
| `METRIC_SESSION_DURATION` | `photon.security.session.duration` |
| `ATTR_LOGIN_RESULT` | `photon.security.login.result` (the `ELoginResult` constant name — bounded) |
| `ATTR_LOGIN_USER_ID` | `photon.security.login.user.id` (**span only**, and only when `isAnonymousLogging ()` is `false`) |
| `UNIT_LOGIN` | `{login}` |
| `UNIT_USER` | `{user}` |
| `UNIT_MILLIS` | `ms` |

Plus a `SecurityMetrics` instrument holder and a package-private `LoggedInUserTelemetry` emitter, mirroring `ph-oton-mgrs/.../longrun/`.

- **Counter** `photon.security.login.failed`, attribute `result` = `eLoginResult.name ()`.
  Note: `ELoginResult` implements `ICredentialValidationResult` (`ISuccessIndicator` + `IHasDisplayText`) and has **no** `getID ()`. Use `name ()` — do not use `getDisplayText (...)`, which is localized and therefore not a stable metric dimension.
- **Counter** `photon.security.login.success`, attribute `result` (distinguishes `SUCCESS` from `SUCCESS_WITH_LOGOUT`).
- **Counter** `photon.security.logout` — only on the `EChange.CHANGED` path.
- **Gauge** `photon.security.users.loggedin` over `getLoggedInUserCount ()`.
- **Histogram** `photon.security.session.duration` in ms, recorded in `logoutUser` from `Duration.between (aInfo.getLoginDT (), aInfo.getLogoutDT ())`, which the method already computes for its log line. No attributes.

## Gauge lifecycle caveat

`TelemetryMetrics.gauge (...)` returns an `ITelemetryGauge` that is `Closeable`. `LoggedInUserManager` is an `AbstractGlobalSingleton`, so the gauge must be created when the singleton is instantiated and closed in its destruction hook — **not** held in a `static final` field like the counters, or it will outlive the global scope and keep a stale manager alive across a servlet-context restart. The hook is `AbstractSingleton.onDestroy (IScope)` (there is also `onBeforeDestroy (IScope)`); both are `protected` and declare `throws Exception`.

## POM

`ph-oton-security/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Every `ELoginResult` failure value produces exactly one increment, `USER_NOT_EXISTING` included.
- [ ] No user ID, login name or IP address on any metric attribute; span user ID honours `isAnonymousLogging ()`.
- [ ] The gauge is closed when the global singleton is destroyed.
- [ ] `logoutUser`'s "user was not logged in" early return does not count as a logout.
- [ ] Unit test with recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`, asserting one increment per `ELoginResult` failure path.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-security/.../security/CSecurityTelemetry.java` | new, public - metric / attribute names for the whole module |
| `ph-oton-security/.../security/login/LoginMetrics.java` | new, public - 3 counters, 1 histogram, gauge factory |
| `ph-oton-security/.../security/login/LoggedInUserTelemetry.java` | new, package-private emitter |
| `ph-oton-security/.../security/login/LoggedInUserManager.java` | modified |
| `ph-oton-security/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-security/src/test/.../LoggedInUserTelemetryTest.java` | new, 6 tests |
| `ph-oton-security/src/test/.../RecordingTelemetryMeterSPI.java` + service file | new test infrastructure - see below |

### Emission

* Counter `photon.security.login.failed`, attribute `photon.security.login.result` = `eLoginResult.name ()`.
* Counter `photon.security.login.success`, same attribute - so `SUCCESS` and `SUCCESS_WITH_LOGOUT` stay distinguishable.
* Counter `photon.security.logout` - only on the `EChange.CHANGED` path; the "user was not logged in" early return is not counted, and the test asserts that.
* Histogram `photon.security.session.duration` from the `Duration` the method already computes for its log line. The log line was refactored to compute that duration once into a local instead of inline.
* Observable gauge `photon.security.users.loggedin` over `getLoggedInUserCount ()`.

### All failure paths covered

`_onLoginError (...)` is the choke point for `USER_IS_DELETED`, `USER_IS_DISABLED`, `INVALID_PASSWORD`, `USER_IS_MISSING_ROLE`, `SESSION_ALREADY_HAS_USER` and `USER_ALREADY_LOGGED_IN` - both from `loginUser (IUser, ...)` and from `internalSessionActivateUser (...)`. One increment there covers all of them.

`USER_NOT_EXISTING` is returned **twice** without going through that choke point, and both sites got their own increment:

* `loginUser (String sLoginName, ...)` when the login name resolves to no user, and
* `loginUser (IUser aUser, ...)` when `aUser` is `null`.

The first returns directly rather than delegating to the second, so there is exactly one increment per call - no double counting. Both are asserted by separate tests.

### Gauge lifecycle

As required, the gauge is **not** a `static final` field. It is created in `onAfterInstantiation (IScope)` and closed in `onDestroy (IScope)` of the global singleton:

```java
// Must not be a static field - it would outlive the global scope
private ITelemetryGauge m_aLoggedInUsersGauge;
```

`LoginMetrics.createLoggedInUsersGauge (LongSupplier)` is a factory method rather than a constant, and its Javadoc says why.

### Two deviations from the issue text

**1. `ATTR_LOGIN_USER_ID` was not added.** The constant was specified as span-only, honouring `isAnonymousLogging ()`. But `LoggedInUserManager` starts no span, and ph-telemetry has no "current span" accessor - the issue itself ruled out driving an API addition there. A span-only attribute constant with no use site would be dead code, so it is left out; `isAnonymousLogging ()` consequently never comes into play. The privacy rule still holds and is asserted: the failure counter has exactly **one** attribute, the login result.

**2. `internalSessionActivateUser (...)` also counts its success.** The issue only named the tail of `loginUser (IUser, ...)`. But that method's failures already increment via `_onLoginError`, and its success puts the user into `m_aLoggedInUsers` and fires `onUserLogin` exactly like a real login. Counting only its failures would have made the success counter disagree with the gauge and with the logout counter. It emits `result=SUCCESS`.

### Test infrastructure - relevant for the remaining tiers

The acceptance criteria asked for `Telemetry.install (...)` / `TelemetryMetrics.install (...)` with a reset in `@After`. **That does not work in this module** and the reason generalises: the instruments in `LoginMetrics` are resolved once in its static initializer, and `LoggedInUserManagerTest` sorts before `LoggedInUserTelemetryTest`, logs users in, and would bind the instruments to the no-op meter before the telemetry test gets a chance to install anything.

The recording meter is therefore registered through the regular SPI mechanism instead - `src/test/resources/META-INF/services/com.helger.telemetry.ITelemetryMeterSPI` - so it is picked up whenever the instruments are first created, regardless of test order. `RecordingTelemetryMeterSPI` collects into static state and the tests clear it in `@Before`. As a bonus this is the only place in the repo that exercises the real SPI discovery path. `SPITest` passes with the new service file.

### Verification

Whole ph-oton-security suite: 49/49 green, including `SPITest` and the pre-existing `LoggedInUserManagerTest`. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.5.1 - work in progress`.
